File: app/src/main/java/starbright/com/projectegg/features/userAccount/UserAccountContract.java
package starbright.com.projectegg.features.userAccount;

import starbright.com.projectegg.features.base.BasePresenter;
import starbright.com.projectegg.features.base.BaseView;

class UserAccountContract {

    interface View extends BaseView<Presenter> {
        void updateView(boolean isLogin);

        void navigateToSearchRecipePage();

        void disableView();

        void enableView();

        void setupProgressBar();
        void showProgressBar();
        void hideProgressBar();

        void showEmailEmptyErrorToast();
        void showEmailFormatWrongErrorToast();
        void showPasswordErrorToast();

        void showLoginErrorToast(String errorMessage);

        void showVerificationEmailSentDialog();
    }

    interface Presenter extends BasePresenter {
        void handleVerificationEmailDialogClicked();
        void handleAuthenticationButtonClicked(String email, String password);

        void updateUserAuthRole();
        void onDestroy();

        boolean isLoginAuthentication();
    }
}

File: app/src/main/java/starbright/com/projectegg/features/userAccount/UserAccountContract.kt
/*
 * Copyright (c) by Andreas (oentoro.andreas@gmail.com)
 * created at 25 - 7 - 2020.
 */

/**
 * Created by Andreas on 7/10/2019.
 */

package starbright.com.projectegg.features.userAccount

class UserAccountContract {

    interface View {
        fun updateView(isLogin: Boolean)

        fun navigateToSearchRecipePage()

        fun disableView()

        fun enableView()

        fun setupProgressBar()

        fun showProgressBar()

        fun hideProgressBar()

        fun showEmailEmptyErrorToast()

        fun showEmailFormatWrongErrorToast()

        fun showPasswordErrorToast()

        fun showLoginErrorToast(errorMessage: String?)

        fun showVerificationEmailSentDialog()
    }

    interface Presenter {

        fun handleVerificationEmailDialogClicked()

        fun handleAuthenticationButtonClicked(email: String, password: String)

        fun handleOnBackPressedClicked(): Boolean

        fun updateUserAuthRole()

        fun onDestroy()
    }
}
